### DISCUSSION TITLE
Latest LeafMap version that works with numpy 1.24.4?

### DISCUSSION THREAD

**Discussion by nve-sak** [opengeos/leafmap#1010] [Q&A] [answered]
Hello, I am moving a project from development on Windows to deployment under a somwhat old Ubuntu, and running trough a lot of required downgrades. As I am through the list, there remains a conflict in LeafMap 0.37.6 requiring numpy >= 1.25.0, but the latest available numpy is 1.24.4. How far do I have to downgrade LeafMap?

**Comment by giswqs** [comment, marked as answer]
Leafmap does not pin numpy version. The problem is that numpy>1.24.4 depends on distutils, which was  removed from the standard library in Python 3.12. Therefore, you won't able to able to build numpy if distutils does not exist in your env. 

You can use [uv](https://docs.astral.sh/uv/) to install Python 3.11 and leafmap. 

```bash
uv venv --python 3.11 
uv pip install leafmap numpy==1.24.4
```

**Reply by nve-sak** [reply]
Thanks :-) I went to python 3.13, and the entire requirements.txt from Windows installed successfully on Ubuntu at the first attempt. Got another problem, though, but unrelated, so I created a new question #1011.
